mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -231,6 +238,38 @@ def _azure_upload_file(self, credentials, local_file, artifact_path):
         except Exception as err:
             raise MlflowException(err)
 
+    def _azure_adls_gen2_upload_file(self, credentials, local_file, artifact_path):
+        """
+        Uploads a file to a given Azure storage location using the ADLS gen2 API.
+        """
+        try:
+            headers = self._extract_headers_from_credentials(credentials.headers)
+
+            # try to create the file
+            try:
+                put_adls_file_creation(credentials.signed_uri, headers=headers)
+            except requests.HTTPError as e:
```

Comment:
We need to catch auth errors and refresh the presigned URL if an auth error is encountered, as in https://github.com/mlflow/mlflow/blob/af348a12d42bacb6b62b0b97cf931412dc45aa85/mlflow/store/artifact/databricks_artifact_repo.py#L225-L232

Reply:
I see.  I originally misread what this block was for...will reintroduce.